https.c

```diff
@@ -571,10 +570,16 @@ https_send(struct https_request *req, const char *method, const char *uri,
                    ctx->useragent);
 
         /* Add headers */
-        if (hdrs != NULL && (p = strdup(hdrs)) != NULL) {
-                BIO_printf(req->cbio, "%s\r\n", strtok(p, "\r\n"));
-                while ((p = strtok(NULL, "\r\n")) != NULL && *p)
-                        BIO_printf(req->cbio, "%s\r\n", p);
+        if (hdrs != NULL) {
+                char *hdrs_dup = strdup(hdrs);
+                if (hdrs_dup != NULL) {
+                        char *p = strtok(hdrs_dup, "\r\n");
```

Comment:
My C days are far behind me.  Do we need to free this pointer too?

Reply:
No, strtok() modifies hdrs_dup. After executing this line p will be equal to hdrs_dup.